sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects.scala

```diff
@@ -439,6 +439,13 @@ case class MapObjects(
       s"boolean ${loopVar.isNull} = ${genInputData.isNull} || ${loopVar.value} == null;"
     }
 
+    // If lambdaFunction is WrapOption, we will not determine null or not based on the
+    // value of loopVar.isNull, because WrapOption will return None for null.
+    val isWrapOption = lambdaFunction match {
```

Comment:
I'm not sure exactly what is going on here, but this looks like a hack.  Ideally this node should not need to know what type its child is to operate correctly.